### PR TITLE
Fix Fiber devtools prod

### PR DESCRIPTION
**Pull request by spicyj**

### CODE REVIEW

File: src/renderers/shared/fiber/ReactFiberDevToolsHook.js [hunk 1]
```diff
@@ -12,15 +12,13 @@
...
+var warning = require('warning');
+
 import type { Fiber } from 'ReactFiber';
 import type { FiberRoot } from 'ReactFiberRoot';
 
 declare var __REACT_DEVTOOLS_GLOBAL_HOOK__ : Object | void;
 
-if (__DEV__) {
-  var warning = require('warning');
-}
```
Comment (iamdustan) [inline src/renderers/shared/fiber/ReactFiberDevToolsHook.js]: slight alternative: it looks like 2 of the 3 `warning()` call sites in here are wrapped in `__DEV__`. Would it make sense to wrap the 3rd one in `__DEV__` as well, or unwrap all of them?

Comment (gaearon) [inline src/renderers/shared/fiber/ReactFiberDevToolsHook.js]: I'll just break it again later this way 😄

**Comment by spicyj** [comment]
I couldn't write a test to break this because we transform `warning(a)` to `a && warning(false)` in open source but I added a test anyway in case we somehow have another regression in this path.

**Pull request merged**